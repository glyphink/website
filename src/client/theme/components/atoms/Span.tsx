/**
 * @file Atoms - Span
 * @module website/components/atoms/Span
 */

import type { Props } from '@glyphink/website/theme'
import {
  forwardRef,
  type ForwardedRef,
  type ForwardRefExoticComponent,
  type JSX
} from 'react'
import type SpanProps from './Span.props.mts'

/**
 * A generic inline container for phrasing content.
 *
 * @see https://developer.mozilla.org/docs/Web/HTML/Reference/Elements/span
 * @see {@linkcode ForwardRefExoticComponent}
 * @see {@linkcode SpanProps}
 *
 * @category
 *  atoms
 *
 * @const {ForwardRefExoticComponent<SpanProps>} Span
 */
const Span: ForwardRefExoticComponent<SpanProps> = forwardRef<
  HTMLSpanElement,
  SpanProps
>(
  /**
   * @param {Props<SpanProps>} props
   *  Component props
   * @param {ForwardedRef<HTMLSpanElement>} ref
   *  Forwarded ref
   * @return {JSX.Element}
   *  JSX `<span>` element
   */
  (
    props: Props<SpanProps>,
    ref: ForwardedRef<HTMLSpanElement>
  ): JSX.Element => {
    return <span {...props} ref={ref} />
  }
)

Span.displayName = 'Span'

export default Span
