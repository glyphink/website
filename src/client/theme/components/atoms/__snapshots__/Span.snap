// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:theme/components/atoms/Span > should render <span> 1`] = `
<span>
  The quick brown fox jumped over the lazy dog
</span>
`;
